subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaAnnotationProcessingIntegrationTest.groovy

```diff
@@ -267,4 +285,50 @@ class JavaAnnotationProcessingIntegrationTest extends AbstractIntegrationSpec {
         then:
         succeeds "compileJava"
     }
+
+    def "wraps processing in build operation"() {
```

Comment:
I'm think this test only checks in-process compilation. We should have coverage for fork=true as well.

Reply:
Good catch! Added in f1ecfc146c4.